### PR TITLE
Revamp previous native implementation

### PR DESCRIPTION
**Pull request by dulvinw**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 1]
```diff
@@ -144,6 +144,9 @@ public BuildCommand(Path userDir, PrintStream outStream, PrintStream errStream,
     @CommandLine.Option(names = "--dump-llvm-ir", hidden = true)
     private boolean dumpLLVMIR;
 
+    @CommandLine.Option(names = "--no-optimize-llvm", hidden = true)
+    private boolean noOptimizeLlvm;
+
```
Comment (KavinduZoysa) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: shall we change variable name as below?
noOptimizeLlvm -> noOptimizeLLVM

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java [hunk 2]
```diff
@@ -42,16 +42,26 @@
 public class CreateJarTask implements Task {
 
     private boolean dumpBir;
+    private boolean buildNative;
+    private boolean dumpLlvmIr;
+    private boolean noOptimizeLlvm;
```
Comment (KavinduZoysa) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java]: shall we change the variable names as,
dumpLlvmIr -> dumpLLLVMIr
noOptimizeLlvm -> noOptimizeLLVM

File: compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_gen.bal [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+function optimize(llvm:LLVMModuleRef mod) {
+    var passBuilder = llvm:llvmPassManagerBuilderCreate();
+    llvm:llvmPassManagerBuilderSetOptLevel(passBuilder, 3);
+
+    var pass = llvm:llvmCreatePassManager();
+    llvm:llvmPassManagerBuilderPopulateFunctionPassManager(passBuilder, pass);
+    llvm:llvmPassManagerBuilderPopulateModulePassManager(passBuilder, pass);
+    // comment above two lines and uncomment below when debuing
+    //llvm:LLVMAddPromoteMemoryToRegisterPass(pass);
+
```
Comment (KavinduZoysa) [inline compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_gen.bal]: Please remove this comment.

File: compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_func_gen.bal [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/llvm;
+//import ballerina/io;
+import ballerina/bir;
```
Comment (KavinduZoysa) [inline compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_func_gen.bal]: Shall we remove  this unused import.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (manuranga) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: https://stackoverflow.com/questions/15526107/acronyms-in-camelcase

File: compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_bb_body_gen.bal [hunk 5]
```diff
@@ -29,17 +37,14 @@ type BbBodyGenrator object {
...
-            panic err;
         }
     }
 
     function genMoveIns(bir:Move moveIns) {
         llvm:LLVMValueRef lhsRef = self.parent.getLocalVarRefById(moveIns.lhsOp.variableDcl.name.value);
         var rhsVarOp = moveIns.rhsOp;
         llvm:LLVMValueRef rhsVarOpRef = self.parent.genLoadLocalToTempVar(rhsVarOp);
-        var loaded = llvm:LLVMBuildStore(self.builder, rhsVarOpRef, lhsRef);
+        var loaded = <llvm:LLVMValueRef> llvm:llvmBuildStore(self.builder, rhsVarOpRef, lhsRef);
```
Comment (dulvinw) [inline compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_bb_body_gen.bal]: Check if the cast is needed.

Comment (dulvinw) [reply compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_bb_body_gen.bal]: Refactoring in a seperate PR

File: compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_func_gen.bal (hunk 4 above)
Comment (dulvinw) [reply compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_func_gen.bal]: Refactoring in a seperate PR

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java (hunk 2 above)
Comment (dulvinw) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java]: Refactoring in a seperate PR

File: compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_gen.bal (hunk 3 above)
Comment (dulvinw) [reply compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_gen.bal]: Refactoring in a seperate PR

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (dulvinw) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Refactoring in a seperate PR

**Pull request merged**